FWCore/Integration/test/EDAlias_t.cpp

```diff
@@ -116,3 +117,241 @@ process.moduleToTest(process.test, cms.Task(process.intprod))
+)
+
+process.intalias = cms.EDAlias(intprod = cms.VPSet(cms.PSet(type = cms.string('*'), fromProductInstance = cms.string('another'))))
+
+# Module to be tested can not be an EDAlias
+process.test = cms.EDProducer('AddIntsProducer',
+    labels = cms.VInputTag({})
+)
+
+process.moduleToTest(process.test, cms.Task(process.intprod))
+)_"};
+
+  edm::test::TestProcessor::Config config{fmt::format(baseConfig, "'intalias:another'")};
+
+  SECTION("Base configuration is OK") { REQUIRE_NOTHROW(edm::test::TestProcessor(config)); }
+
+  SECTION("No event data") {
+    edm::test::TestProcessor tester(config);
+    REQUIRE_NOTHROW(tester.test());
+  }
+
+  SECTION("beginJob and endJob only") {
+    edm::test::TestProcessor tester(config);
+    REQUIRE_NOTHROW(tester.testBeginAndEndJobOnly());
+  }
+
+  SECTION("Run with no LuminosityBlocks") {
+    edm::test::TestProcessor tester(config);
+    REQUIRE_NOTHROW(tester.testRunWithNoLuminosityBlocks());
+  }
+
+  SECTION("LuminosityBlock with no Events") {
+    edm::test::TestProcessor tester(config);
+    REQUIRE_NOTHROW(tester.testLuminosityBlockWithNoEvents());
+  }
+
+  SECTION("Getting value") {
+    edm::test::TestProcessor tester(config);
+    auto event = tester.test();
+    REQUIRE(event.get<edmtest::IntProduct>()->value == 4);
+  }
+
+  SECTION("Other product instances are not aliased") {
+    {
+      edm::test::TestProcessor::Config config{fmt::format(baseConfig, "'intalias:foo'")};
+      edm::test::TestProcessor tester(config);
+      REQUIRE_THROWS_WITH(tester.test(), Catch::Contains("ProductNotFound"));
+    }
+    {
+      edm::test::TestProcessor::Config config{fmt::format(baseConfig, "'intalias'")};
+      edm::test::TestProcessor tester(config);
+      REQUIRE_THROWS_WITH(tester.test(), Catch::Contains("ProductNotFound"));
+    }
+  }
+}
+
+////////////////////////////////////////
+TEST_CASE("Configuration with all products of two modules", s_tag) {
+  const std::string baseConfig{
+      R"_(from FWCore.TestProcessor.TestProcess import *
+import FWCore.ParameterSet.Config as cms
+
+process = TestProcess()
+
+process.intprod = cms.EDProducer('ManyIntProducer', ivalue = cms.int32(2),
+    values = cms.VPSet(
+        cms.PSet(instance=cms.string('foo'),value=cms.int32(3)),
+        cms.PSet(instance=cms.string('another'),value=cms.int32(4)),
+    )
+)
+process.intprod2 = cms.EDProducer('ManyIntProducer', ivalue = cms.int32(20),
+    values = cms.VPSet(
+        cms.PSet(instance=cms.string('foo2'),value=cms.int32(30)),
+        cms.PSet(instance=cms.string('another2'),value=cms.int32(40)),
+    )
+)
+
+process.intalias = cms.EDAlias(
+    intprod = cms.EDAlias.allProducts(),
+    # can't use allProducts() because the product instance '' would lead to duplicate brances to be aliased
+    intprod2 = cms.VPSet(
+        cms.PSet(type = cms.string('*'), fromProductInstance = cms.string('foo2')),
+        cms.PSet(type = cms.string('*'), fromProductInstance = cms.string('another2')),
+    )
+)
```

Comment:
does this mean that 
```python
alias = cms.EDAlias(
    module = cms.VPSet(
        cms.PSet(
            type = cms.string("*")
        )
    )
)
```
and the `allProducts()` shorthand work only if a module does not produce multiple collections with the same type ?

Reply:
> does this mean that ... and the `allProducts()` shorthand work only if a module does not produce multiple collections with the same type ?

It does not. The first added test
https://github.com/cms-sw/cmssw/blob/c0144f29ad5860ab6eef8d62fe71e7b15fa0967d/FWCore/Integration/test/EDAlias_t.cpp#L136
tests exactly that, creating an EDAlias for a module that produces three `edmtest::IntProduct` collections.

This test tests a case where products from two modules are aliased-for in a single EDAlias, and that type type wildcard works there as well. Because of my laziness of constructing the test (essentially the `intprod` and `intprod2` producing products of the same type, and both producing a product without product instance name) the simpler
```cpp
process.intalias = cms.EDAlias(
    intprod = cms.EDAlias.allProducts(),
    intprod2 = cms.EDAlias.allProducts(),
)
```
would lead to two products of `(edmtest::IntProduct, "intalias", "", "TEST")`, which is not allowed.

Reply:
Ah, I see, the conflict would be between two modules producing a collection with the same type *and* label.